src/lib.rs

```diff
@@ -310,19 +322,19 @@ impl<R: gimli::Reader> Context<R> {
             .take_while(move |i| {
                 // We know that this CU's start is beneath the probe already because
                 // of our sorted array.
-                debug_assert!(i.range.begin <= probe);
+                debug_assert!(i.range.begin <= probe_high);
 
                 // Each entry keeps track of the maximum end address seen so far,
                 // starting from the beginning of the array of unit ranges. We're
                 // iterating in reverse so if our probe is beyond the maximum range
                 // of this entry, then it's guaranteed to not fit in any prior
                 // entries, so we break out.
-                probe <= i.max_end
+                probe_low < i.max_end
             })
             .filter_map(move |i| {
                 // If this CU doesn't actually contain this address, move to the
                 // next CU.
-                if probe > i.range.end {
+                if probe_low > i.range.end || probe_high <= i.range.begin {
```

Comment:
Should this be `probe_low >= i.range.end`?

Reply:
I was replicating the original logic - does that mean the original should be `probe >= i.range.end`? (I suspect so.)

Reply:
No I think that's correct, but I got `probe_low <= i.max_end` wrong a couple of lines above.

Reply:
A reason I commented is because this check and the one a couple of lines above need to be consistent, unless my understanding is wrong. They are consistent on master, but this PR made them inconsistent, and 6001cecaf504044c4b9c906f877cd3decee2140e changes them both again so they are still inconsistent. I think `probe_low >= i.range.end` is correct here, because `i.range.end` is exclusive (and `i.max_end` is exclusive too), meaning this was a bug on master. So you need to change the one a couple of lines above again to be `probe_low < i.max_end`.